### PR TITLE
Add files for alt34 keyboard design

### PR DESCRIPTION
**Pull request by altosys**
This adds configuration files and a default keymap for my custom keyboard design `alt34`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please review the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and verify that you have followed the checklist for the type of PR this is.

**Comment by altosys** [comment]
> Please review the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and verify that you have followed the checklist for the type of PR this is.

I have gone through the checklist as well as I can, probably just forgot to check the box.

**Comment by altosys** [comment]
@drashna I'm not entirely sure what you meant with "links in make file", but otherwise I think I got the issues you mentioned sorted out.

File: keyboards/alt34/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = no           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/alt34/rules.mk:16]:
```suggestion
NKRO_ENABLE = no           # Enable N-Key Rollover
```

File: keyboards/alt34/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# MCU name
```
Comment (drashna) [inline keyboards/alt34/rules.mk:1]: Also, the content could be moved to the rev1 folder, and would remove the need some of the includes you have.

File: keyboards/alt34/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
Comment (drashna) [inline keyboards/alt34/rules.mk:9]:
```suggestion
#   change to "no" to disable the options
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/alt34/readme.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# alt34
+
+![alt34](https://i.imgur.com/GDv0dL0.png)
```
Comment (drashna) [inline keyboards/alt34/readme.md:3]:
```suggestion
![alt34](https://i.imgur.com/GDv0dL0h.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/alt34/readme.md (hunk 4 above)
Comment (altosys) [reply keyboards/alt34/readme.md:3]: I'm not that familiar with the fine details of imgur, what does adding `h` to the end of the filename do?

File: keyboards/alt34/rules.mk (hunk 2 above)
Comment (altosys) [reply keyboards/alt34/rules.mk:1]: You mean that the entire contents of my `alt34/rules.mk` could be moved to the `alt34/rev1/rules.mk`?

Comment (drashna) [reply keyboards/alt34/rules.mk:1]: correct.  And no, actually, youd want to remove .t

File: keyboards/alt34/rev1/rules.mk [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# Build Options
+#   change to "no" to disable the options
+
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = no           # Enable N-Key Rollover
+UNICODE_ENABLE = no        # Unicode
```
Comment (fauxpark) [inline keyboards/alt34/rev1/rules.mk:16]:
```suggestion
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
`.noci` file can be deleted.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by altosys** [comment]
Wonderful!

**Pull request merged**